src/commands/synthetics/utils/internal.ts

```diff
@@ -87,3 +88,97 @@ export const toExecutionRule = (env: string | undefined): ExecutionRule | undefi
+type AccumulatorBaseConfigOverride = Omit<
+  BaseConfigOverride,
+  'basicAuth' | 'headers' | 'cookies' | 'deviceIds' | 'locations' | 'tunnel' | 'variables' | 'retry' // These options will be implemented later in separate PRs, see https://datadoghq.atlassian.net/browse/SYNTH-12971
+> & {
+  'retry.count'?: number
+  'retry.interval'?: number
+}
+type AccumulatorBaseConfigOverrideKeys = keyof AccumulatorBaseConfigOverride
+type OverridesValues = boolean | number | string | string[] | ExecutionRule
+
+export const parseOverrideValue = (value: string, type: validTestOverridesValues): OverridesValues => {
+  switch (type) {
+    case 'boolean':
+      const parsedBoolean = toBoolean(value)
+      if (parsedBoolean !== undefined) {
+        return parsedBoolean
+      }
+      throw new Error(`Invalid boolean value: ${value}`)
+    case 'number':
+      const parsedNumber = toNumber(value)
+      if (parsedNumber !== undefined) {
+        return parsedNumber
+      }
+      throw new Error(`Invalid number value: ${value}`)
+    case 'string':
+      return value.trim()
+    case 'string[]':
+      return value.split(';').map((item) => item.trim())
+    case 'ExecutionRule':
+      const parsedExecutionRule = toExecutionRule(value)
+      if (parsedExecutionRule !== undefined) {
+        return parsedExecutionRule
+      }
+      throw new Error(`Invalid ExecutionRule value: ${value}`)
+    default:
+      throw new Error(`Unknown type: ${type}`)
+  }
+}
+
+export const validateAndParseOverrides = (overrides: string[] | undefined): AccumulatorBaseConfigOverride => {
+  if (!overrides) {
+    return {}
+  }
+
+  return overrides.reduce((acc: AccumulatorBaseConfigOverride, override: string) => {
+    const [key, value] = override.split('=') as [AccumulatorBaseConfigOverrideKeys, string]
+
+    // here, we want to parse the value according to its type. With sepcial case for retry
+    switch (key) {
+      // Convert to numbers
+      case 'defaultStepTimeout':
+      case 'pollingTimeout':
+      case 'retry.count':
+      case 'retry.interval':
+      case 'testTimeout':
+        acc[key] = parseOverrideValue(value, 'number') as number
+        break
+
+      // Convert to strings
+      case 'body':
+      case 'bodyType':
+      case 'startUrl':
+      case 'startUrlSubstitutionRegex':
+        acc[key] = parseOverrideValue(value, 'string') as string
+        break
+
+      // Convert to boolean
+      case 'allowInsecureCertificates':
+      case 'followRedirects':
+        acc[key] = parseOverrideValue(value, 'boolean') as boolean
+        break
+
+      // Convert to ExecutionRule
+      case 'executionRule':
+        acc[key] = parseOverrideValue(value, 'ExecutionRule') as ExecutionRule
+        break
+
+      // Special parsing for resourceUrlSubstitutionRegexes
+      case 'resourceUrlSubstitutionRegexes':
+        acc['resourceUrlSubstitutionRegexes'] = acc['resourceUrlSubstitutionRegexes'] ?? []
```

Comment:
```suggestion
        acc['resourceUrlSubstitutionRegexes'] ??= []
```

Reply:
Hmm it seems like the linter doesn't support the nullish assignment

Reply:
Ok, forget about it then, it was only syntax sugar 😁